scripts/airflow/tasks/build_artery_centreline/build_artery_centreline.sql

```diff
@@ -1,26 +1,42 @@
-drop table if exists prj_volume.artery_centreline;
-create table prj_volume.artery_centreline as (
-  (
-    select
-      pvat.arterycode,
-      pvat.centreline_id,
-      1 as centreline_type,
-      ST_ClosestPoint(gc.geom, ST_Centroid(gc.geom)) as geom
-    from gis.centreline gc
-    join prj_volume.artery_tcl pvat
-    on gc.geo_id = pvat.centreline_id
-    where pvat.artery_type = 1 and pvat.centreline_id is not null
-    order by pvat.arterycode asc
-  ) UNION ALL (
-    select
-      arterycode,
-      cast(int_id as bigint) as centreline_id,
-      2 as centreline_type,
-      ST_SetSrid(ST_MakePoint(longitude, latitude), 4326) as geom
-    from prj_volume.artery_intersections
+CREATE SCHEMA IF NOT EXISTS prj_volume;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS prj_volume.artery_centreline AS (
+  WITH artery_centreline_raw AS (
+    (
+      select
+        pvat.arterycode,
+        pvat.centreline_id,
+        1 as centreline_type,
+        ST_Transform(ST_ClosestPoint(
+          ST_Transform(gc.geom, 2952),
+          ST_Transform(ST_Centroid(gc.geom), 2952)
+        ), 4326) as geom
+      from gis.centreline gc
+      join prj_volume.artery_tcl pvat
+      on gc.geo_id = pvat.centreline_id
+      where pvat.artery_type = 1 and pvat.centreline_id is not null
+      order by pvat.arterycode asc
+    ) UNION ALL (
+      select
+        arterycode,
+        cast(int_id as bigint) as centreline_id,
+        2 as centreline_type,
+        ST_SetSrid(ST_MakePoint(longitude, latitude), 4326) as geom
+      from prj_volume.artery_intersections
+    )
   )
+  SELECT
+    arterycode,
+    centreline_id,
+    centreline_type,
+    MODE() WITHIN GROUP(ORDER BY geom) AS geom
```

Comment:
we should really be using `MODE()` for collisions normalization as well.  (doesn't need to be fixed right away, just flagging it here.)